src/Materials/material.ts

```diff
@@ -661,6 +661,10 @@ export class Material implements IAnimatable {
     constructor(name: string, scene: Scene, doNotAdd?: boolean) {
         this.name = name;
         this.id = name || Tools.RandomId();
+        let idSubscript = 0;
+        while (scene.getMaterialByID(this.id) != null) {
+            this.id = name + "_" + idSubscript;

```

Comment:
You probably wanted to do `this.id = name + "_" + (idSubscript++);`

Also you may use `!==` instead of `!=`.

Reply:
I would recommend removing `!= null` completely.